src/topology.c

```diff
@@ -7,9 +7,8 @@
  *
  */
 
-#include <hwloc.h>
-
 #include "base_alloc_global.h"
+#include "provider_os_memory_internal.h"
```

Comment:
please move do not create fake dependencies between "objects" - this file do not need os_memory_provider. it needs hwloc

Reply:
done